tools/autodeployment/server.js

```diff
@@ -53,35 +58,50 @@ function handleLog(req, res) {
   out.on('end', () => end('Build Complete.'));
 }
 
-const handleGitPush = createHandler({ path: '/', secret: SECRET });
+const handleGitPush = new Webhooks({ path: '/', secret: SECRET });
 
-https
-  .createServer(credentials, (req, res) => {
-    // Build Status Info as JSON
-    if (req.url === '/status') {
-      handleStatus(req, res);
-    }
-    // Build Log Stream (if build is happening)
-    else if (req.url === '/log') {
-      handleLog(req, res);
-    }
-    // Process GitHub Push Event, or error (404)
-    else {
-      handleGitPush(req, res, () => handleError(req, res));
-    }
-  })
-  .listen(DEPLOY_PORT, () => {
-    console.log(`Server listening on port ${DEPLOY_PORT}.\nUse /status or /log for build info.`);
-  });
+// Build Status Info as JSON
+app.get('/status', function (req, res) {
+  handleStatus(req, res);
+});
 
-handleGitPush.on('error', (err) => {
-  console.error('Error:', err.message);
+// Build Log Stream (if build is happening)
+app.get('/log', function (req, res) {
+  handleLog(req, res);
 });
 
-if (!(DEPLOY_TYPE === 'staging' || DEPLOY_TYPE === 'production')) {
-  console.error("DEPLOY_TYPE must be one of 'staging' or 'production'");
-  process.exit(1);
-}
+// Process GitHub Push Event
+app.post('/', async function (req, res) {
+  try {
+    /**
+     * @octokit/webhooks .verifyAndReceive
+     * https://github.com/octokit/webhooks.js#webhooksverifyandreceive
+     */
+    await handleGitPush.verifyAndReceive({
+      id: req.header('X-GitHub-Hook-ID'),
+      name: req.header('X-GitHub-Event'),
+      payload: req.body,
+      signature: req.header('X-Hub-Signature'),
+    });
+
+    handleStatus(req, res);
+  } catch ({ errors }) {
+    handleError(req, res, errors);
+  }
+});
+
+// Error (404)
+app.get('*', function (req, res) {
+  handleError(req, res);
+});
+
+/*
+ * @octokit/webhooks .onError
+ * https://github.com/octokit/webhooks.js#webhooksonerror
+ */
+handleGitPush.onError((err) => {
```

Comment:
We should turn off the `Avoid callbacks. Prefer Async/Await` eslint error, since using callbacks is fine.

Reply:
@humphd globally, or just for this case?